### PR TITLE
KAFKA-17185: Declare Loggers as static to prevent multiple logger instances

### PR DESCRIPTION
**Pull request by mingyen066**
As discussed in https://github.com/apache/kafka/pull/16657#discussion_r1686938593 , we should make logger as static to avoid creating multiple logger instances.
I use the regex `private.*Logger.*LoggerFactory` to search and check all the results if certain logs need to be static. 

There are some exceptions that loggers don't need to be static:
* The logger in the inner class. Since java8 doesn't support static field in the inner class.
  * https://github.com/apache/kafka/blob/trunk/clients/src/test/java/org/apache/kafka/clients/consumer/internals/FetchRequestManagerTest.java#L3676
* Custom loggers for each instance (non-static + non-final). In this case, multiple logger instances is actually really needed.
  * https://github.com/apache/kafka/blob/trunk/storage/src/test/java/org/apache/kafka/server/log/remote/storage/LocalTieredStorage.java#L166
* The logger is initialized in constructor by LogContext. Many non-static but with final modifier loggers are in this category, that's why I use `.*LoggerFactory` to only check the loggers that are assigned initial value when declaration.
* `protected final Logger log = Logger.getLogger(getClass())`
This is for subclass can do logging with subclass name instead of superclass name.
But in this case, if the log access modifier is private, the purpose cannot be achieved since subclass cannot access the log defined in superclass. So if access modifier is private, we can replace `getClass()` with `<className>.class`

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: clients/src/main/java/org/apache/kafka/clients/ClusterConnectionStates.java [hunk 1]
```diff
@@ -42,7 +42,7 @@ final class ClusterConnectionStates {
     static final int CONNECTION_SETUP_TIMEOUT_EXP_BASE = 2;
     static final double CONNECTION_SETUP_TIMEOUT_JITTER = 0.2;
     private final Map<String, NodeConnectionState> nodeState;
-    private final Logger log;
+    private static final Logger log;
```
Comment (chia7712) [inline clients/src/main/java/org/apache/kafka/clients/ClusterConnectionStates.java]: Noted that this static variable should NOT be initialized in constructor.

Comment (mingyen066) [reply clients/src/main/java/org/apache/kafka/clients/ClusterConnectionStates.java]: Thanks for finding the bug. This is the case that I shouldn't change the logger to static.
I have completed the code updates with different approach and also written the methodology and criteria used to PR message. PTAL.

**Comment by chia7712** [comment]
@mingyen066 Could you please merge trunk to run CI again? there are thread leak. I don't think there are related to your PR, but it is always good to check it again

**Comment by mingyen066** [comment]
@chia7712 Sure, I merged trunk branch and the CI is running again.

**Review by chia7712 — APPROVED** [review]
LGTM

**Pull request merged**